### ISSUE TITLE
How to add images on disk to gpx, so that QMapShack will handle them?

### ISSUE THREAD

**Issue by googol42** [Maproom/qmapshack#425]
This is not a bug report, but a question to which I haven't found the answer.

### Is your feature request related to a problem? Please describe.
I want to generate a gpx file by a python program and I want to attach images to waypoints so that QMapShack will display the image when the waypoint is viewed or edited.

What I have tried to far:
- search the internet
- checked an exported gpx file (with extensions) -> There are no references to the images on disc.
- I had a look at QMapShack's database and did not find any references to my files on disc. I had the assumption that only an id is saved to gpx and that the path to the image is only kept in QMapShack's database. But this didn't hold true.
- I search the source code, but didn't find any useful information. (However, I found out that images properly linked in QMapShack can be handled by Garmin which makes me want this even more!) I had to directions: 1) The image might be added to a database/cache/whatever when adding an image, but I got stuck here https://github.com/Maproom/qmapshack/blob/3df55d4aebb1e323f31c4ba4c138dfc218fc33e4/src/qmapshack/widgets/CPhotoAlbum.cpp#L111 (which component listens to the signal?). 2) When serializing the data to a gpx. What happens with images? https://github.com/Maproom/qmapshack/blob/3df55d4aebb1e323f31c4ba4c138dfc218fc33e4/src/qmapshack/gis/gpx/CGpxProject.cpp#L320

My current guess is, that it is the `<ql:key>...</ql:key>` contains the key needed for the linking.

### Describe the solution you'd like
An explanation how QMapShack manages images. How can I add my own images 

### Describe alternatives you've considered
There are none.

### Give a reason why someone else should spend time on that request
I did my homework and tried quite a few things. Adding images is a very nice feature. People using QMapShack. Would be good, when other people could find an answer to this question. I could publish part of my python script here in the issue, once I figured out how things work.

### Additional context
-

**Comment by googol42** [comment]
I just figured I looked at the wrong database. I now looked at the `workspace.db` and there is my gpx file. Is the image (path) contained in the data column in the table "workspace"?

**Comment by kiozen** [comment]
That's right, images are only supported by QMapShack's binary format and they are not stored in full resolution. So it's probably not  what you want. 

GPX does not have any support for images. You could link to files with extended tags. However that is not portable either as other software would need to support those special tags, too.

**Comment by googol42** [comment]
What sql would I need to execute to add the images myself?

**Comment by kiozen** [comment]
They are serialized together with all other item data in a binary blob. See `quint64 CDBProject::insertItem(IGisItem* item, QSqlQuery& query)` for details on how items are stored in the database. In line 426 the item data gets serialized by serializing the complete history of the item. 

So it's not that straight forward as updating the database entry by a simple sql command.
